server.go

```diff
@@ -8,174 +8,203 @@ import (
 
 // Server is a go-socket.io server.
 type Server struct {
+	engine *engineio.Server
+
 	handlers map[string]*namespaceHandler
-	eio      *engineio.Server
 }
 
 // NewServer returns a server.
-func NewServer(c *engineio.Options) (*Server, error) {
-	eio, err := engineio.NewServer(c)
-	if err != nil {
-		return nil, err
-	}
+func NewServer(c *engineio.Options) *Server {
+	engine, _ := engineio.NewServer(c)
+
 	return &Server{
 		handlers: make(map[string]*namespaceHandler),
-		eio:      eio,
-	}, nil
+		engine:   engine,
+	}
 }
 
 // Close closes server.
 func (s *Server) Close() error {
-	return s.eio.Close()
+	return s.engine.Close()
 }
 
 func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
-	s.eio.ServeHTTP(w, r)
+	s.engine.ServeHTTP(w, r)
 }
 
 // OnConnect set a handler function f to handle open event for
 // namespace nsp.
 func (s *Server) OnConnect(nsp string, f func(Conn) error) {
-	h := s.getNamespace(nsp, true)
+	h := s.getNamespace(nsp)
+	if h == nil {
+		h = s.createNameSpace(nsp)
+	}
+
 	h.OnConnect(f)
 }
 
 // OnDisconnect set a handler function f to handle disconnect event for
 // namespace nsp.
 func (s *Server) OnDisconnect(nsp string, f func(Conn, string)) {
-	h := s.getNamespace(nsp, true)
+	h := s.getNamespace(nsp)
+	if h == nil {
+		h = s.createNameSpace(nsp)
+	}
+
 	h.OnDisconnect(f)
 }
 
 // OnError set a handler function f to handle error for namespace nsp.
 func (s *Server) OnError(nsp string, f func(Conn, error)) {
-	h := s.getNamespace(nsp, true)
+	h := s.getNamespace(nsp)
+	if h == nil {
+		h = s.createNameSpace(nsp)
+	}
+
 	h.OnError(f)
 }
 
 // OnEvent set a handler function f to handle event for namespace nsp.
 func (s *Server) OnEvent(nsp, event string, f interface{}) {
-	h := s.getNamespace(nsp, true)
+	h := s.getNamespace(nsp)
+	if h == nil {
+		h = s.createNameSpace(nsp)
+	}
+
 	h.OnEvent(event, f)
 }
 
 // Serve serves go-socket.io server
 func (s *Server) Serve() error {
 	for {
-		conn, err := s.eio.Accept()
+		conn, err := s.engine.Accept()
+		//todo maybe need check EOF from Accept()
```

Comment:
`todo` here too? Fix, remove, or document more what needs to be done?

Reply:
i will fix to it this release